src/Avalonia.Base/Media/GlyphTypeface.cs

```diff
@@ -0,0 +1,407 @@
+                        }
+
+                        var culture = GetCulture(nameRecord.LanguageID);
+
+                        faceNames ??= new Dictionary<CultureInfo, string>(1);
+
+                        if (!faceNames.ContainsKey(culture))
+                        {
+                            faceNames[culture] = nameRecord.Value;
+                        }
+                    }
+                }
+
+                FamilyNames = familyNames ?? s_emptyStringDictionary;
+                FaceNames = faceNames ?? s_emptyStringDictionary;
+            }
+            else
+            {
+                FamilyNames = new Dictionary<CultureInfo, string> { { CultureInfo.InvariantCulture, FamilyName } };
+                FaceNames = new Dictionary<CultureInfo, string> { { CultureInfo.InvariantCulture, Weight.ToString() } };
+            }
+
+            static CultureInfo GetCulture(int lcid)
+            {
+                if (lcid == ushort.MaxValue)
+                {
+                    return CultureInfo.InvariantCulture;
+                }
+
+                try
+                {
+                    return CultureInfo.GetCultureInfo(lcid) ?? CultureInfo.InvariantCulture;
+                }
+                catch (CultureNotFoundException)
+                {
+                    return CultureInfo.InvariantCulture;
+                }
+            }
+        }
+
+        public string TypographicFamilyName { get; }
+
+        public IReadOnlyDictionary<CultureInfo, string> FamilyNames { get; }
+
+        public IReadOnlyDictionary<CultureInfo, string> FaceNames { get; }
+
+        public IReadOnlyList<OpenTypeTag> SupportedFeatures
+        {
+            get
+            {
+                if (_supportedFeatures != null)
+                {
+                    return _supportedFeatures;
+                }
+
+                var gPosFeatures = FeatureListTable.LoadGPos(this);
+                var gSubFeatures = FeatureListTable.LoadGSub(this);
+
+                var supportedFeatures = new List<OpenTypeTag>(gPosFeatures?.Features.Count ?? 0 + gSubFeatures?.Features.Count ?? 0);
+
+                if (gPosFeatures != null)
+                {
+                    foreach (var gPosFeature in gPosFeatures.Features)
+                    {
+                        if (supportedFeatures.Contains(gPosFeature))
+                        {
+                            continue;
+                        }
+
+                        supportedFeatures.Add(gPosFeature);
+                    }
+                }
+
+                if (gSubFeatures != null)
+                {
+                    foreach (var gSubFeature in gSubFeatures.Features)
+                    {
+                        if (supportedFeatures.Contains(gSubFeature))
+                        {
+                            continue;
+                        }
+
+                        supportedFeatures.Add(gSubFeature);
+                    }
+                }
```

Comment:
Nit: extract to a `LoadSupportedFeatures()` method and simplify this property to `_supportedFeatures ??= LoadSupportedFeatures()` for readability.